vimiv/imutils/exif.py

```diff
@@ -10,139 +10,271 @@
+        exif = dict()
+
+        try:
+
+            for ifd in self._metadata:
+                if ifd == "thumbnail":
+                    continue
+
+                for tag in self._metadata[ifd]:
+                    keyname = piexif.TAGS[ifd][tag]["name"]
+                    keytype = piexif.TAGS[ifd][tag]["type"]
+                    val = self._metadata[ifd][tag]
+                    _logger.debug(
+                        f"name: {keyname}\
+                        type: {keytype}\
+                        value: {val}\
+                        tag: {tag}"
+                    )
+                    if keyname not in desired_keys:
+                        _logger.debug(f"Ignoring key {keyname}")
+                        continue
+                    if keytype in (
+                        piexif.TYPES.Byte,
+                        piexif.TYPES.Short,
+                        piexif.TYPES.Long,
+                        piexif.TYPES.SByte,
+                        piexif.TYPES.SShort,
+                        piexif.TYPES.SLong,
+                        piexif.TYPES.Float,
+                        piexif.TYPES.DFloat,
+                    ):  # integer and float
+                        exif[keyname] = (keyname, val)
+                    elif keytype in (
+                        piexif.TYPES.Ascii,
+                        piexif.TYPES.Undefined,
+                    ):  # byte encoded
+                        exif[keyname] = (keyname, val.decode())
+                    elif keytype in (
+                        piexif.TYPES.Rational,
+                        piexif.TYPES.SRational,
+                    ):  # (int, int) <=> numerator, denominator
+                        exif[keyname] = (keyname, f"{val[0]}/{val[1]}")
+
+        except (piexif.InvalidImageDataError, KeyError):
+            return None
+
+        return exif
+
+    def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
+
+        try:
+            if reset_orientation:
+                with contextlib.suppress(KeyError):
+                    self._metadata["0th"][
+                        piexif.ImageIFD.Orientation
+                    ] = ExifOrientation.Normal
+            exif_bytes = piexif.dump(self._metadata)
+            piexif.insert(exif_bytes, dest)
+            _logger.debug("Successfully wrote exif data for '%s'", dest)
+        except piexif.InvalidImageDataError:  # File is not a jpg
+            _logger.debug("File format for '%s' does not support exif", dest)
+        except ValueError:
+            _logger.debug("No exif data in '%s'", dest)
+
+    def exif_date_time(self) -> str:
+
+        with contextlib.suppress(
+            piexif.InvalidImageDataError, FileNotFoundError, KeyError
+        ):
+            return self._metadata["0th"][piexif.ImageIFD.DateTime].decode()
+        return ""
+
+
+class _ExifHandlerNoExif(_ExifHandler):
+    """ExifHandler implementation for no exif support."""
+
+    def __init__(self, filename=""):
```

Comment:
I get a linting error here because `filename` is unused. Can I simple add a `# pylint: disable=unused-argument` to ignore it it is there a better solution?

Reply:
You could also just prepend it with an underscore, i.e. `_filename`, to indicate that it is unused.

Reply:
Thanks, fixed in c3feef5f119e2ebd3190ce87d7e0d567745429c5